gt/main.c

```diff
@@ -1447,10 +1445,30 @@ run_gt(struct net_config *net_conf, struct gt_config *gt_conf)
 	if (gt_conf->num_lcores <= 0)
 		goto success;
```

Comment:
I don't remember what scenario would cause the GT block to be run but to have 0 lcores allocated to it, or why we consider this a "success" event. Do we still need this scenario?

And if we do, should the allocation of the strings below also be in this scenario (under the `success` label below)?

Reply:
Good point! I will make sure the allocation successfully in this scenario.

Reply:
Another point is that: if we allocate the resources even we have 0 lcores allocated to GT, the question is when should we release these resources? It seems the only place to call ``gt_conf_put()`` is in ``gt_proc()``.

Reply:
We can deal `gt_conf->num_lcores <= 0` as an error and avoid the issue of calling `gt_conf_put()` that @mengxiang0811 found.

The comment that @mengxiang0811 references is correct, but it was written in a very early version of Gatekeeper. At that time, we didn't have the lua scripts that initialize Gatekeeper organized in such a way that either GK or GT blocks are called. Anyway, thank you for uncovering that reference!

Do analogous change to `gk_run()`. This two patches (i.e. one for GT and other for GK) could come in another, small pull request that we could quickly merge.